### PR TITLE
DEVOPS-452: Update curve-apps package with python-poetry-template

### PR DESCRIPTION
**Pull request by SophieCurinier**
**DEVOPS-452 - apply latest version of python template repos to all repos**
- Use reusable github workflows
- Use MiraGeoscience/pre-commit-hooks package in `.pre-commit-hooks.yml`
- Update pre-commit-hooks version:
   - `ruff-pre-commit` : v0.5.5
   - `mirrors-mypy`: v10.0.0 -> v1.10.1
   - `Mirageoscience/pre-commit-hooks`: v1.0.0
   - `codespell`: v2.2.6 -> v2.3.0
   - `rstcheck`: v6.2.0 -> v6.2.4

### CODE REVIEW

File: .github/workflows/python_analysis.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+jobs:
+  call-workflow-static-analysis:
+    name: Static analysis
+    uses: MiraGeoscience/CI-tools/.github/workflows/reusable-python-static_analysis.yml@main
+    with:
+      package_manager: 'conda'
+      app_name: ${{ github.env.app_name }}
+      python_vers: '3.10'
+      shell_type: 'bash -l {0}'
```
Comment (sebhmg) [inline .github/workflows/python_analysis.yml:36]: **questison:** is shell_type required?

as the commands within the shared workflow itself depends on the shell type, it does not seem right if the calling workflow has to specify it

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -16,26 +16,17 @@ repos:
...
--   repo: https://github.com/PyCQA/flake8
-    rev: 7.0.0
-    hooks:
-    -   id: flake8
--   repo: https://github.com/asottile/pyupgrade
-    rev: v3.15.2
-    hooks:
-    -   id: pyupgrade
-        args: [--py310-plus]
+    -   id: ruff
```
Comment (sebhmg) [inline .pre-commit-config.yaml:22]: **issue:** the ruff configuration must be provided in pyproject.toml

Comment (sebhmg) [reply .pre-commit-config.yaml:22]: (and config for isort and black should be removed from pyproject.toml)

**Review by sebhmg — CHANGES_REQUESTED** [review]
- misses the Ruff configuration in pyproject.toml
- it feels wrong the job has to specify the shell for the shared workflow

File: .github/workflows/python_analysis.yml (hunk 1 above)
Comment (SophieCurinier) [reply .github/workflows/python_analysis.yml:36]: Yes, it is. Indeed, between poetry and conda, the shell type is not the same.
Within shared workflows, we can always add a step defining the given shell_type, whether it's windows pytest or unix pytest, and whether it's a poetry or conda package.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (SophieCurinier) [reply .pre-commit-config.yaml:22]: It's done. ✅

Comment (SophieCurinier) [reply .pre-commit-config.yaml:22]: ✅

**Review by sebhmg — APPROVED** [review]

**Review by sebhmg — APPROVED** [review]

File: curve_apps-assets/__init__.py [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+#                                                                                '
+#  All rights reserved.                                                          '
+#                                                                                '
+#  This file is part of curve-apps.                                              '
+#                                                                                '
+#  curve-apps is distributed under the terms and conditions of the MIT License   '
+#  (see LICENSE file at the root of this source code package).                   '
+#  '''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+# pylint: disable=invalid-name
```
[the remark this answers is not in the document]
Comment (sebhmg) [reply curve_apps-assets/__init__.py:12]: nevermind, can be fixed up later

**Comment by SophieCurinier** [comment]
@sebhmg could you upload the new rules-set ? (or give me the right to do it 😉 )

**Comment by sebhmg** [comment]
> @sebhmg could you upload the new rules-set ? (or give me the right to do it 😉 )

@SophieCurinier, I think you have the permission now.  I will call you just to go through it together once.

**Pull request merged**